daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -297,6 +305,17 @@ private[lf] final class Compiler(
   private[this] def compileCommands(cmds: ImmArray[Command]): t.SExpr =
     pipeline(translateCommands(Env.Empty, cmds))
 
+  private[this] def compileWithContractDisclosures(
+      sexpr: t.SExpr,
+      disclosures: ImmArray[DisclosedContract],
+  ): t.SExpr = {
+    val disclosureLambda = pipeline(
+      translateContractDisclosureLambda(Env.Empty, disclosures)
+    )
+
+    t.SEApp(disclosureLambda, Array(sexpr))
```

Comment:
FIXME: not convinced that this is the correct thing to do here as `SEApp` expands to `SEAppGeneral` forcing the body to be evaluated first (in a context where disclosures have yet to be evaluated)